qt_gui/src/qt_gui/plugin_manager.py

```diff
@@ -318,8 +318,11 @@ def _shutdown_plugin(self, instance_id, callback):
         # shutdown plugin before unloading
         info = self._running_plugins[str(instance_id)]
         handler = info['handler']
-        handler.close_signal.disconnect(self.unload_plugin)
-        handler.shutdown_plugin(callback)
+        try:
+            handler.close_signal.disconnect(self.unload_plugin)
+            handler.shutdown_plugin(callback)
+        except TypeError:
```

Comment:
Can you please provide more information in which cases both calls can raise a `TypeError` and why it should be handled differently from other exceptions.